configuration/configuration.go

```diff
@@ -202,6 +202,9 @@ type Configuration struct {
 				Deny []string `yaml:"deny,omitempty"`
 			} `yaml:"urls,omitempty"`
 		} `yaml:"manifests,omitempty"`
+		// RepositoryClass validation checks that the class of the configurations inside the
+		// the manifest matches the authorized resource when provided.
+		RepositoryClass bool `yaml:"repositoryclass,omitempty"`
```

Comment:
This should be a list.

Also, this is not a validation, it is an authorization/policy option. May have to think about how to integrate that correctly.